src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetWebApi2Integration.cs

```diff
@@ -122,6 +122,12 @@ public static class AspNetWebApi2Integration
                 }
                 catch (Exception ex)
                 {
+                    if (scope != null)
```

Comment:
Zach, is `UpdateSpan` of concern too? I see that it is called from `ExecuteAsyncInternal`.

Reply:
The `UpdateSpan` method has a try/catch block encapsulating all of the operations of extracting info and placing it in the span so I'm not worried about it throwing any exceptions. Is that what you're asking about?

Reply:
Yes, thanks for looking at that!